packages/webview_flutter/webview_flutter_wkwebview/test/src/web_kit_webview_widget_test.dart

```diff
@@ -303,7 +303,53 @@ void main() {
           verify(mockWebView.setCustomUserAgent('myUserAgent'));
         });
 
-        testWidgets('zoomEnabled', (WidgetTester tester) async {
+        testWidgets('change zoomEnabled to true', (WidgetTester tester) async {
+          when(mockWebViewWidgetProxy.createScriptMessageHandler()).thenReturn(
+            MockWKScriptMessageHandler(),
+          );
+
+          await buildWidget(
+            tester,
+            creationParams: CreationParams(
+              webSettings: WebSettings(
+                userAgent: const WebSetting<String?>.absent(),
+                zoomEnabled: false,
+                hasNavigationDelegate: false,
+              ),
+              javascriptChannelNames: <String>{'myChannel'},
+            ),
+          );
+
+          clearInteractions(mockUserContentController);
+
+          await testController.updateSettings(WebSettings(
+            userAgent: const WebSetting<String?>.absent(),
+            zoomEnabled: true,
+          ));
+
+          final List<dynamic> javaScriptChannels = verifyInOrder(<Object>[
```

Comment:
I think this is non-obvious enough how this relates to the test (as described in the test name) to warrant a comment saying that it's validating that the scripts that aren't related to zoom are re-added.

Or even better: split this into two tests, one which just verifies that the zoom script isn't re-add, and the other called 'enabling zoom preserves other scripts' or something along those lines, which does the `javascriptChannelName` setup and script-add validation.